### PR TITLE
Multisite support

### PR DESCRIPTION
**Pull request by acatton**

### CODE REVIEW

**Comment by rockymeza** [comment]
yay

File: widgy/contrib/widgy_mezzanine/models.py [hunk 1]
```diff
@@ -91,3 +131,63 @@ class Meta:
...
+        Length of a step for the materialized tree
+        """
+        return self.lhs.field.model.steplen
+
+    def as_sqlite(self, qn, connection):
+        lhs, params = qn.compile(self.lhs)
+        return 'SUBSTR(%s, %%s, %%s)' % lhs, (params + [0, self.steplen+1])
+
+    def as_mysql(self, qn, connection):
+        raise NotImplementedError
```
Comment (gavinwahl) [inline widgy/contrib/widgy_mezzanine/models.py]: Widgy supports all the databases Django does.

File: widgy/contrib/widgy_mezzanine/models.py [hunk 2]
```diff
@@ -62,6 +69,39 @@ class PageSelectRelatedManager(SelectRelatedManager, PageManager):
     """
     use_for_related_fields = True
 
+    def get_widgy_owners(self, item):
+        user = current_request().user
```
Comment (gavinwahl) [inline widgy/contrib/widgy_mezzanine/models.py]: Why are you doing this? I'd expect any use of a hack like this to have a darn good explanation.

**Comment by gavinwahl** [comment]
Why are there two sitepermission rows: 

image

**Comment by gavinwahl** [comment]
Merge #340 and rebase on master so travis can run your tests.

File: widgy/models/versioning.py [hunk 3]
```diff
@@ -193,9 +193,18 @@ def get_owner_related_names(cls):
 
     @cached_property
     def owners(self):
-        return list(owner
-                    for attr in self.get_owner_related_names()
-                    for owner in getattr(self, attr).all())
+        acc = []
+        for attr in self.get_owner_related_names():
+            manager = getattr(self, attr)
```
Comment (gavinwahl) [inline widgy/models/versioning.py]: Did you try using _base_manager here to get all objects?

File: widgy/models/versioning.py [hunk 4]
```diff
@@ -193,9 +193,18 @@ def get_owner_related_names(cls):
...
+
+            # If the manager has a get_widgy_owners() method, use this.
```
Comment (gavinwahl) [inline widgy/models/versioning.py]: This comment is obvious. Explain why you're doing what you're doing, not what you're doing.

File: widgy/contrib/widgy_mezzanine/models.py [hunk 5]
```diff
@@ -91,3 +131,63 @@ class Meta:
 
 
 links.register(Link)
+
+
+class PathRoot(models.Transform):
```
Comment (gavinwahl) [inline widgy/contrib/widgy_mezzanine/models.py]: Needs unit tests

File: widgy/contrib/widgy_mezzanine/admin.py [hunk 6]
```diff
@@ -25,7 +27,10 @@
 from widgy.forms import WidgyFormMixin, VersionedWidgyWidget
 from widgy.contrib.widgy_mezzanine import get_widgypage_model
 from widgy.contrib.widgy_mezzanine.views import ClonePageView, UnpublishView
+from widgy.contrib.form_builder.admin import FormAdmin
+from widgy.contrib.form_builder.models import Form
 from widgy.db.fields import get_site
+from widgy.models import Node, VersionTracker
```
Comment (gavinwahl) [inline widgy/contrib/widgy_mezzanine/admin.py]: VersionTracker is imported but unused

File: widgy/models/versioning.py (hunk 3 above)
Comment (acatton) [inline widgy/models/versioning.py]: I dunno what it means.

Comment (gavinwahl) [inline widgy/models/versioning.py]: It looks like you're trying to bypass CurrentSiteManager. I'd expect _base_manager to be used for that and have the filtering happen somewhere else. I don't think it's appropriate to have owners not actually return all owners.

**Pull request merged**